factory.go

```diff
@@ -37,11 +37,23 @@ func (f factory) Create(config *Config, logger log.Logger) (deployer.Connector,
 	if err != nil {
 		return &Connector{}, fmt.Errorf("python binary check failed with error: %w", err)
 	}
-	python := cliwrapper.NewCliWrapper(pythonPath, config.WorkDir, logger)
+
+	workdir, err := os.MkdirTemp(filepath.Clean(config.WorkDir), "")
+	if err != nil {
+		return nil, fmt.Errorf(
+			"error creating temporary directory for python connector (%w)", err)
+	}
```

Comment:
Does this change mean that it always needs to pull, and if there are two of the same type they both end up being pulled separately?
If that's the case, one possible design would be to clone them as done before, then copy them locally to the temp dir.

Reply:
That's fair, but you would need to refactor the design so that the connector knows if this is the first pull request. I think it's a good idea, but probably worth a different issue.

Reply:
But the way it was designed before meant that the directory naming is consistent for each hash you pull. So if it matches, no need to pull. You would, however, need to synchronize to ensure that nothing does anything when another is already pulling.